perl.h

```diff
@@ -720,13 +720,61 @@ Now a placeholder that declares nothing
 =for apidoc AmnUu|void|STMT_START
 =for apidoc_item ||STMT_END
 
-This allows a series of statements in a macro to be used as a single statement,
+These allow a series of statements in a macro to be used as a single statement,
 as in
 
  if (x) STMT_START { ... } STMT_END else ...
 
-Note that you can't return a value out of them, which limits their utility.
-If you need that, instead use C language C<static inline> functions.
+Note that you can't return a value out of this construct and cannot use it as
+an operand to the comma operator, which limits its utility.  If you need
+those features, instead use C language C<static inline> functions.
+Nevertheless, this construct can be quite helpful, and is frequently used.
+
+Subtle bugs can occur due to this being a macro with names evaluated in
+the context of the caller, unlike a function.  These bugs can be eliminated by
+treating this as a function, using only variables passed as parameters or
+declared within the construct.  Otherwise there is the potential for namespace
+collisions with the caller.  For a concrete example, see
+L<https://perlmonks.org/?node_id=11144355>.
+
+Further, all parameters should have an internal version declared and
```

Comment:
I found this sentence confusing a few times, until I realised we weren't talking about a version number.

Reply:
@hvds, I've redone the documentation.  Please review